crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -856,6 +856,88 @@ impl OlmMachine {
         }
     }
 
+    async fn receive_to_device_event(
+        &self,
+        changes: &mut Changes,
+        mut raw_event: Raw<AnyToDeviceEvent>,
+    ) -> OlmResult<Raw<AnyToDeviceEvent>> {
+        let event: ToDeviceEvents = match raw_event.deserialize_as() {
+            Ok(e) => e,
+            Err(e) => {
+                // Skip invalid events.
+                warn!("Received an invalid to-device event: {e}");
+
+                return Ok(raw_event);
+            }
+        };
+
+        trace!(
+            sender = event.sender().as_str(),
+            event_type = %event.event_type(),
```

Comment:
How about adding a span to the new function via `#[instrument]` and add `sender` and `event_type` as fields on the span so that ever tracing event in this function (except the `warn!`) will have them logged? Can be done the same way I did it [here](https://github.com/matrix-org/matrix-rust-sdk/commit/46fe998a33b4c8fb0f9623aafa439d765446d02e).

Reply:
Another span or modify the one on `receive_to_device_event()`?

Reply:
Oh, I only saw the first commit so far 😅

The idea would be to just add two more fields to the existing span then.